tools/Makefile

```diff
@@ -1,14 +1,29 @@
-all:	directory-cache
+.PHONY: check-password
 
-authority_certificate:
-	@echo "Please copy authority_* files to this directory"
-	exit 1
+AUTH_DIR=authority-private
+DIR_CACHE=directory-cache
 
-directory-cache: authority_certificate pip_install
-	mkdir -p directory-cache
-	./gen_fresh_dirinfo.py generate-dirinfo --consensus directory-cache/consensus.txt --microdescriptors directory-cache/microdescriptors.txt
+all: directory-cache
+
+directory-cache: pip_install check-password
+	mkdir -p $(DIR_CACHE)
+	mkdir -p $(AUTH_DIR)
+	./gen_fresh_dirinfo.py generate-certificate \
+	--authority-identity-key $(AUTH_DIR)/authority_identity_key \
+	--authority-signing-key $(AUTH_DIR)/authority_signing_key \
+	--authority-certificate $(DIR_CACHE)/certificate.txt \
+	--authority-v3ident $(DIR_CACHE)/authority.txt
+	./gen_fresh_dirinfo.py generate-dirinfo \
+	--authority-signing-key $(AUTH_DIR)/authority_signing_key \
+	--authority-certificate $(DIR_CACHE)/certificate.txt \
+	--consensus $(DIR_CACHE)/consensus.txt \
+	--microdescriptors $(DIR_CACHE)/microdescriptors.txt
 
 pip_install: requirements.txt
 	pip3 install -qr requirements.txt
 	touch pip_install
 
+check-password:
+ifeq ($(DIR_AUTH_PASSWORD), )
+	$(error environment variable DIR_AUTH_PASSWORD is undefined or invalid); fi
+endif
```

Comment:
this is evaluated at load time, not when calling `check-password`, probably not what you want. and there is a dangling `fi` also

Reply:
Oops! Indeed, that is not what I intended...